Lista/ListaDuplamenteEncadeada.c

```diff
@@ -104,117 +78,116 @@ boolean lista_inserir_fim(LISTA *lista, ITEM *item)
         if (pnovo != NULL)
         {
             pnovo->item = item;
-            if (lista->inicio == NULL)
+            if (lista->sentinela->proximo == NULL)
             {
-                lista->inicio = pnovo;
-                pnovo->anterior = NULL;
+                pnovo->anterior = pnovo->proximo = NULL;
+                lista->sentinela->proximo = pnovo;
             }
             else
             {
                 lista->fim->proximo = pnovo;
                 pnovo->anterior = lista->fim;
+                pnovo->proximo = lista->sentinela;
+                if (lista->fim->anterior == NULL)
+                {
+                    lista->fim->anterior = pnovo;
+                }
             }
-            pnovo->proximo = NULL;
             lista->fim = pnovo;
             lista->tamanho++;
             return TRUE;
         }
     }
     return FALSE;
 }
-
-boolean lista_remover_item(LISTA *lista, int chave)
+ITEM *busca(const LISTA *lista, int chave)
 {
-    NODE *noAtual = NULL;
-    if (lista_contem_algo(lista))
-    {
-        noAtual = lista->inicio;
-        /*Percorre a lista em busca da chave*/
-        noAtual = getNodeWithKey(chave, noAtual);
-        /*Se a lista não acabou significa que encontrou a chave*/
-        deleteNodeIfPossible(lista, noAtual);
-    }
-    return FALSE; /*elemento (chave) não está na lista ou lista vazia*/
-}
+    lista->sentinela->item = item_criar_vazio();

```

Comment:
não sei bem como tá implementado isso, mas se isso aqui for um malloc e tal, a gnt vai ficar criando 1 item sempre que for pesquisar, mas não vai apagar ele não?
acho que pra isso, dá pra a gnt dar um free nesse lista->sentinela->item no final da busca que tá de boa

Reply:
boa, eu tive que colocar isso pq na busca não fazia sentido vc coloca item_set_chave(lista->sentinela->item), se sentinela não era um item hauhauah